angular/src/components/modal/dynamic-modal.component.ts

```diff
@@ -0,0 +1,57 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    Component,
+    ComponentFactoryResolver,
+    ComponentRef,
+    ElementRef,
+    OnDestroy,
+    Type,
+    ViewChild,
+    ViewContainerRef
+} from '@angular/core';
+
+import { ModalRef } from './modal.ref';
+
+@Component({
+    selector: 'app-modal',
+    template: '<ng-template #modalContent></ng-template>',
```

Comment:
This is neat, I like the way the base html for this is just a subtle ng-template.